kubelet/datadog_checks/kubelet/kubelet.py

```diff
@@ -97,12 +102,21 @@ def get_config(key):
 
 DEPRECATED_SUMMARIES = {
     'kubelet_network_plugin_operations_latency_microseconds': 'kubelet.network_plugin.latency',
+    'kubelet_pod_start_latency_microseconds': 'kubelet.pod.start.duration',
+    'kubelet_pod_worker_latency_microseconds': 'kubelet.pod.worker.duration',
+    'kubelet_pod_worker_start_latency_microseconds': 'kubelet.pod.worker.start.duration',
+    'kubelet_runtime_operations_latency_microseconds': 'kubelet.runtime.operations.duration',
+    'kubelet_docker_operations_latency_microseconds': 'kubelet.docker.operations.duration',
```

Comment:
should we mark the deprecated metrics in `metadata.csv`?

Reply:
In fact, those metrics have been deprecated in Kubernetes 1.14+:
https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/CHANGELOG-1.14.md#deprecated-metrics
but they have been replaced by new ones.
I added the new ones below in the `TRANSFORM_VALUE_HISTOGRAM` map.
The old and the new kubernetes metrics are generating the same datadog metrics.
The unit is different. That’s why it goes through a transformer.

For ex., in https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/CHANGELOG-1.14.md#deprecated-metrics
> * kubelet_pod_start_latency_microseconds -> kubelet_pod_start_duration_seconds

For this metric, we have in this check:
```python
DEPRECATED_SUMMARIES = {
    'kubelet_pod_start_latency_microseconds': 'kubelet.pod.start.duration',
    …

    TRANSFORM_VALUE_HISTOGRAMS = {
        'kubelet_pod_start_duration_seconds': 'kubelet.pod.start.duration',
```

So, the check will always generate a `kubelet.pod.start.duration` even if the “source” metric changes between old and new versions of kubernetes.

In the `metadata.csv` file, we document the Datadog metric. And from the Datadog side, it’s not deprecated.